frontend/src/metabase/entities/containers/EntityListLoader.jsx

```diff
@@ -159,20 +169,52 @@ export default class EntityListLoader extends React.Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    const { keepPreviousList } = this.props;
+    const { previousList } = this.state;
+    if (!keepPreviousList) {
+      return;
+    }
+
+    if (
+      Array.isArray(prevProps.list) &&
+      !_.isEqual(previousList, prevProps.list)
+    ) {
+      this.setState({ previousList: prevProps.list });
+    }
+  }
+
   renderChildren = () => {
-    let { children, entityDef, wrapped, list, reload, ...props } = this.props; // eslint-disable-line no-unused-vars
+    let {
+      children,
+      entityDef,
+      wrapped,
+      list,
+      reload, // eslint-disable-line no-unused-vars
```

Comment:
If we don't use vars, maybe delete the line here.

Reply:
TLDR: it's there because it should not be passed to a composed component

`EntityListLoader` is a HOC. It wraps another component and passes some extra props extending the component's functionality. Also, it should still pass down the props that are required by a wrapped component.

`reload` is a boolean prop configuring `EntityListLoader` behavior and it should not be passed down to the wrapped component. All the other HOC props are removed with `_.omit` [here](https://github.com/metabase/metabase/blob/47280d84f713bb9930ba4800a66747bcccf7aa4b/frontend/src/metabase/entities/containers/EntityListLoader.jsx#L204). However, it's intentionally removed from the list of props to omit [here](https://github.com/metabase/metabase/blob/47280d84f713bb9930ba4800a66747bcccf7aa4b/frontend/src/metabase/entities/containers/EntityListLoader.jsx#L52). Looks like the idea is that the list loader also passes `reload` function as prop and it should overwrite the boolean `reload` props (see [here](https://github.com/metabase/metabase/blob/47280d84f713bb9930ba4800a66747bcccf7aa4b/frontend/src/metabase/entities/containers/EntityListLoader.jsx#L208))

Overall, I guess it's safe to remove the line, but I'm a bit worried because this component is used like a lot

Reply:
Ok, thanks.